ffi/derive/tests/ui_fail/mutable_input_arg.rs

```diff
@@ -0,0 +1,25 @@
+use iroha_ffi::{ffi_bindgen, FfiResult};
+use std::mem::MaybeUninit;
+
+struct FfiStruct {
+    a: u32,
+}
+
+#[ffi_bindgen]
+impl FfiStruct {
```

Comment:
I think this test needs a little explanation. We want to prevent functions which accept mutable arguments from compiling. Do we want that to control side-effects? What kind of UB can you trigger if you **do** disable the part that forbids mutable arguments?

Reply:
it's just that there is no need for supporting mutable input arguments so this simplifies the proc macro implementation by a lot. It's not that it's not possible to support a feature like this, it's unnecessary